Cubical/Algebra/Field/Base.agda

```diff
@@ -0,0 +1,169 @@
+{-# OPTIONS --safe #-}
+module Cubical.Algebra.Field.Base where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.SIP
+open import Cubical.Relation.Nullary
+
+open import Cubical.Data.Sigma
+open import Cubical.Data.Unit
+open import Cubical.Data.Empty
+
+open import Cubical.Algebra.Semigroup
+open import Cubical.Algebra.Monoid
+open import Cubical.Algebra.AbGroup
+open import Cubical.Algebra.Ring
+  hiding (_$_)
+open import Cubical.Algebra.CommRing
+
+open import Cubical.Displayed.Base
+open import Cubical.Displayed.Auto
+open import Cubical.Displayed.Record
+open import Cubical.Displayed.Universe
+
+open import Cubical.Reflection.RecordEquiv
+
+open Iso
+
+private
+  variable
+    ℓ ℓ' : Level
+
+record IsField {R : Type ℓ}
+                  (0r 1r : R) (_+_ _·_ : R → R → R) (-_ : R → R)
+                  (_[_]⁻¹ : (x : R) → ¬ (x ≡ 0r) → R) : Type ℓ where
+
+  constructor isfield
+
+  field
+    isCommRing : IsCommRing 0r 1r _+_ _·_ -_
+    ·⁻¹≡1      : (x : R) (≢0 : ¬ (x ≡ 0r)) → x · (x [ ≢0 ]⁻¹) ≡ 1r
+    0≢1        : ¬ (0r ≡ 1r)
+
+  open IsCommRing isCommRing public
+
+record FieldStr (A : Type ℓ) : Type (ℓ-suc ℓ) where
+
+  constructor fieldstr
+
+  field
+    0r         : A
+    1r         : A
+    _+_        : A → A → A
+    _·_        : A → A → A
+    -_         : A → A
+    _[_]⁻¹     : (x : A) → ¬ (x ≡ 0r) → A
+    isField    : IsField 0r 1r _+_ _·_ -_ _[_]⁻¹
+
+  infix  20 _⁻¹
+  infix  8 -_
+  infixl 7 _·_
+  infixl 7 _/_
+  infixl 6 _+_
+
+  _⁻¹ : (x : A) → { ≢0 : ¬ (x ≡ 0r) } → A
```

Comment:
Wait - this should not be inferrable in practice. The proof that something is non-zero should be explicit.

Reply:
Yes, right, I'm afraid it will not work as intended, so you can just remove it (and the division).